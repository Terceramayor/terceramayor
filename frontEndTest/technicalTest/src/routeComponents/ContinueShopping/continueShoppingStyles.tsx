import { StyleSheet } from 'react-native';
import { FontNames } from '../../../utils/enums';

const dashBoardStyles = StyleSheet.create({
  dashboardContainer: {
    height: '100%',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center'
  },
  products: {
    alignItems: 'center'
  },
  helloUserProfile: {
    marginTop: 50,
    width: '100%',
    paddingTop: 10,
    paddingBottom: 10,
    textAlign: 'center',
    fontFamily: FontNames.MontserratRegular,
    fontSize: 25
  }
});

export default dashBoardStyles;
