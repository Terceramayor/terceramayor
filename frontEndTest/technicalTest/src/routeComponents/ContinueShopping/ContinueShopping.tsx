// eslint-disable-next-line no-use-before-define
import React, { useEffect } from 'react';
import { View, ImageBackground, StyleSheet } from 'react-native';
import Dispatch, { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { increaseDecreaseQuantity } from '../../redux/actions/productsRelatedActions';
import { productCasuistic } from '../../utils/noMagicStrings';
import Item from '../../commonComponents/Item/Item';
import BuyProducts from '../../commonComponents/buyProducts/BuyProducts';
import continueShoppingStyles from './continueShoppingStyles';
import Header from '../../commonComponents/header/Header';

function ContinueShopping({ navigation }) {
  const styles = StyleSheet.create({
    backgroundImage: {

      flex: 1,
      resizeMode: 'cover'
    }
  });
  return (
    <ImageBackground style={styles.backgroundImage} source={require('../../assets/icons/bakcground.png')}>
      <Header />

      <BuyProducts navigation={navigation} />
    </ImageBackground>
  );
}

const mapDispatchToProps = (dispatch) => ({
  actions: bindActionCreators({ increaseDecreaseQuantity }, dispatch)
});

export default connect(null, mapDispatchToProps)(ContinueShopping);
