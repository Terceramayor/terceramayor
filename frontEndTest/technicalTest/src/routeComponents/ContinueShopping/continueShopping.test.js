// eslint-disable-next-line no-use-before-define
import React from 'react';
import { render, fireEvent, act } from '@testing-library/react-native';
import ContinueShopping from './ContinueShopping';

jest.mock('./../../commonComponents/header/Header', () => 'MockedItemComponent');
jest.mock('./../../commonComponents/buyProducts/BuyProducts', () => 'MockedBuyProductsComponent');

describe('Given the ContinueShopping component', () => {
  describe('When it is rendered', () => {
    test('Then the SnapShot should match the rendered tree', () => {
      const navigation = { navigate: jest.fn() };
      jest.spyOn(navigation, 'navigate');

      const tree = render(<ContinueShopping navigation={navigation} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
