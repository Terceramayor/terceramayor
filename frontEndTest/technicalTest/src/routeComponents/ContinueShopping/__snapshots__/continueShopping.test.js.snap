// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ContinueShopping component When it is rendered Then the SnapShot should match the rendered tree 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/bakcground.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <MockedItemComponent />
  <MockedBuyProductsComponent
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
</View>
`;
